zproject/prod_settings_template.py

```diff
@@ -152,6 +152,23 @@
     # "zproject.backends.GenericOpenIdConnectBackend",  # Generic OIDC integration, setup below
 )
 
+## JWT-based users login and API keys fetching.
+##
+## If you want to login users or fetch API keys via JWT in Zulip,
+## you need to set the secret key and algorithm to use to validate
+## JWT tokens received.
+# JWT_AUTH_KEYS: Dict[str, Any] = {
+#     # Subdomain for which you want to enable the service.
+#     "zulip": {
+#         # Shared secret key used to validate jwt tokens. This key
+#         # needs to be securely, randomly generated.
```

Comment:
It would be good to comment more on the parameters for said generation -- e.g. can we say anything about the entropy requirements for this to be secure?

Reply:
Amended this paragraph to mention that:
```
#         # Shared secret key used to validate jwt tokens. This key
#         # needs to be securely, randomly generated. Note that if you're
#         # using the default HS256 algorithm, per RFC 7518, the key needs
#         # to be at least 256 bits long.
```

would it be better to phrase this in terms of bits of entropy rather than key length? (I went with how the RFC talks about it). Also ref https://auth0.com/blog/brute-forcing-hs256-is-possible-the-importance-of-using-strong-keys-to-sign-jwts/ which mentions this requirement, though they say they use 512 bits.

Reply:
Yeah, bits of entropy is the right metric to talk about.

Reply:
Tweaked